java-backend/src/main/java/org/kframework/backend/java/builtins/BuiltinIOOperations.java

```diff
@@ -114,20 +116,24 @@ public static Term parseInModule(StringToken input, StringToken startSymbol, Str
 
     public static Term system(StringToken term, TermContext termContext) {
         Map<String, String> environment = new HashMap<>();
-        String[] args = term.stringValue().split("\001", -1);
-        //for (String c : args) { System.out.println(c); }
+        String[] args = Commandline.translateCommandline(term.stringValue());
         ProcessOutput output = RunProcess.execute(environment, termContext.global().files.getProcessBuilder(), args);
 
-        KLabelConstant klabel = KLabelConstant.of(KORE.KLabel("#systemResult(_,_,_)"), termContext.definition());
-        /*
-        String klabelString = "#systemResult(_,_,_)";
-        KLabelConstant klabel = KLabelConstant.of(klabelString, context);
-        assert def.kLabels().contains(klabel) : "No KLabel in definition for " + klabelString;
-        */
+        KLabelConstant klabel = KLabelConstant.of(KORE.KLabel("#systemResult(_,_,_)_K-IO"), termContext.definition());
         String stdout = output.stdout != null ? new String(output.stdout) : "";
         String stderr = output.stderr != null ? new String(output.stderr) : "";
         return KItem.of(klabel, KList.concatenate(IntToken.of(output.exitCode),
-            StringToken.of(stdout.trim()), StringToken.of(stderr.trim())), termContext.global());
+                StringToken.of(stdout.trim()), StringToken.of(stderr.trim())), termContext.global());
+    }
+
+    public static Term tempFilename(StringToken prefix, StringToken suffix, TermContext termContext) throws IOException {
+        return StringToken.of(File.createTempFile("tmp" + prefix.stringValue(), suffix.stringValue()).getAbsolutePath());
+    }
+
+    public static Term remove(StringToken fname, TermContext termContext) {
+        File f = new File(fname.stringValue());
+        f.delete();
+        return BuiltinList.kSequenceBuilder(termContext.global()).build();
```

Comment:
this needs to return an IOError if the delete operation fails.

Reply:
What about the ocaml-backend? `val remove : string -> unit` returns nothing.

Reply:
on ocaml you should be calling Unix.unlink which raises an exception if there is an error. We already have a function to handle those exceptions; see unix_error in prelude.ml.